packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -574,11 +589,109 @@ export class TransactionAnalyzer {
    * @private
    */
   private validateTransaction() {
-    if (this._originalTx.ins.length === 0) {
+    if (this._rawTx.ins.length === 0) {
       throw new Error("Transaction has no inputs");
     }
-    if (this._originalTx.outs.length === 0) {
+    if (this._rawTx.outs.length === 0) {
       throw new Error("Transaction has no outputs");
     }
   }
+
+  private static validateOptions(
+    options: AnalyzerOptions,
+  ): ValidatedAnalyzerOptions {
+    const validatedOptions: Partial<ValidatedAnalyzerOptions> = {};
+
+    // Raw transaction validation
+    try {
+      const tx = Transaction.fromHex(options.txHex);
```

Comment:
Honestly, you could probably just let the `fromHex` do the validation. bitcoinjs-lib should handle all the important validation. So if it doesn't throw, I'd say that's good enough.

Reply:
Will need to keep the 

```js

    if (tx.outs.length === 0) {
        throw new Error("Transaction has no outputs");
      }
 ```
      
      check , as  bitcoinjs-lib  allow tx with zero outputs